src/compiler/scanner.ts

```diff
@@ -2997,7 +2988,7 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
         }
 
         function isClassContentExit(ch: number) {
-            return ch === CharacterCodes.closeBracket || ch === CharacterCodes.EOF || pos >= end;
+            return ch === CharacterCodes.closeBracket || ch === CharacterCodes.EOF;

```

Comment:
This is unnecessary as all `isClassContentExit` calls are directly after `charCodeChecked`.